metakb/transform/moa.py

```diff
@@ -0,0 +1,489 @@
+
+        proposition_index = self._set_ix(propositions_support_evidence_ix,
+                                         'propositions', key)
+        proposition['_id'] = f"proposition:{proposition_index:03}"
+
+        return [proposition]
+
+    def _get_predicate(self, clin_sig):
+        """Get the predicate of this record
+
+        :param: clinical significance of the assertion
+        :return: predicate
+        """
+        predicate = None
+        if not clin_sig:
+            return None
+        if clin_sig.upper() in schemas.PredictivePredicate.__members__.keys():
+            predicate = schemas.PredictivePredicate[clin_sig.upper()].value
+
+        return predicate
+
+    def _get_variation_origin(self, variant):
+        """Return variant origin.
+
+        :param: A MOA variant record
+        :return: A str representation of variation origin
+        """
+        if variant['feature_type'] == 'somatic_variant':
+            origin = schemas.VariationOrigin.SOMATIC.value
+        elif variant['feature_type'] == 'germline_variant':
+            origin = schemas.VariationOrigin.GERMLINE.value
+        else:
+            origin = None
+
+        return origin
+
+    def _get_variation_descriptors(self, variant, g_descriptors):
+        """Add variation descriptor to therapeutic response
+
+        :param: single evidence(assertion) record from MOA
+        :return: list of variation descriptor
+        """
+        ref_allele_seq = variant['protein_change'][2] \
+            if 'protein_change' in variant and variant['protein_change'] else None  # noqa: E501
+
+        structural_type, molecule_context = None, None
+        if 'variant_annotation' in variant:
+            if variant['variant_annotation'] == 'Missense':
+                structural_type = "SO:0001606"
+                molecule_context = 'protein'
+
+        # For now, the normalizer only support a.a substitution
+        if g_descriptors and 'protein_change' in variant and variant['protein_change']:  # noqa: E501
+            gene = g_descriptors[0]['label']
+            variant_query = f"{gene} {variant['protein_change'][2:]}"
+            validations = self.variant_to_vrs.get_validations(variant_query)
+            v_norm_resp = \
+                self.variant_normalizer.normalize(variant_query,
+                                                  validations,
+                                                  self.amino_acid_cache)
+            vod_value_id = v_norm_resp.value_id if v_norm_resp else None
+            vod_value = v_norm_resp.value if v_norm_resp else None
+        else:
+            vod_value_id = None
+            vod_value = None
+
+        variation_descriptor = schemas.VariationDescriptor(
+            id=f"moa:vid{variant['id']}",
+            label=variant['feature'],
+            value_id=vod_value_id,
+            value=vod_value,
+            gene_context=f"moa:{[g_des['id'] for g_des in g_descriptors]}",
```

Comment:
This is causing the `gene_context` field to have brackets. You should get the first result or return None.